### ISSUE TITLE
Start using labels for Issues and Pull Requests 

### ISSUE THREAD

**Issue by mike-wendt** [rapidsai/cudf#180]
[Numba](https://github.com/numba/numba/issues) uses labels well to tag issues, and as this project and other GoAi projects grow it is hard to easily see where attention is needed on an issue or PR.

For issues, these labels could be helpful for reviewers and contributors to see where they can best help:
* `bug`
* `doc`
* `enhancement` or `feature request`
* `help wanted`
* `question`
* `wontfix`
* `duplicate`
* `proposal` or `discussion`

For PRs these labels can help show the progress of the PR, and for issues this can let the community know where a bug is in the process of being fixed at a glance:
* `0 - Backlog`
* `1 - On Deck`
* `2 - In Progress`
* `3 - Ready for Review`
* `4 - Done`

In addition, we can add labels for specific components like: `cuda`, `python`, etc.

Looking forward to getting input from others and the community on how to best proceed.

**Comment by kkraus14** [comment]
@mike-wendt using labels to help make issues and pull requests easier to manage sounds like a great idea. For the issues labels, I believe we should add tags for focus areas such as `python`, `numba`, `cuda`, `libgdf`, and `arrow` as a start.

I'm unclear as to where you'd use the `4 - Done` label for PRs. After the PR is accepted and merged?

**Comment by mike-wendt** [comment]
@kkraus14 For `4 - Done` I was matching what Numba uses. From what I can see this is their way of flagging a PR as being done with the review process and ready to merge. Happy to reword it to be more clear, like `4 - Ready to Merge` or if we want to notate the outcome of the review have these options: `4 - Review Passed`, `4 - Review Failed` (I guess we should keep them both `4` as it's the same step)

I like the suggested focus areas they are helpful for contributors and others to quickly identify issues/PRs they can assist, based on their own expertise.

**Comment by kkraus14** [comment]
I can understand the `4 - Review Failed` workflow, but if it went to `4 - Review Passed` shouldn't it be merged and that's that?

**Comment by mike-wendt** [comment]
I agree just sometimes someone says `LGTM` or something to that effect indicating the review is complete; however, they are waiting for someone else to do the merge.

**Comment by mike-wendt** [comment]
@scopatz @sklam @seibert @randerzander any thoughts?

**Comment by kkraus14** [comment]
> I agree just sometimes someone says LGTM or something to that effect indicating the review is complete; however, they are waiting for someone else to do the merge.

If someone says LGTM and doesn't have the ability to merge, shouldn't a maintainer still review the PR before merging? Looking through the Numba open/closed PR history I'm not seeing any instances of actually using the `4 - Done` label.

**Comment by seibert** [comment]
We use the process of the reviewer is responsible for the merge.

**Comment by mike-wendt** [comment]
How about the following?
* `0 - Backlog`
* `1 - On Deck`
* `2 - In Progress`
* `3 - Ready for Review`
* `4 - Needs Rework`

**Comment by kkraus14** [comment]
Looks good to me! Thanks for pushing on this @mike-wendt.

**Comment by mike-wendt** [comment]
Labels are live on `libgdf` and here, will leave this up for feedback, and suggestion for missing labels

**Comment by dantegd** [comment]
Are we using both brackets and labels? So the latest issues have both [Bug] and the bug label. Would removing the [bla] after the label is applied be a good idea?

**Comment by mike-wendt** [comment]
@dantegd yes the idea is to remove the `[Bug]` notation in favor of the labels. Just haven't had the time to run through all myself and make the change. Clicking the check box and drop down is about all I had time for today.

**Comment by kkraus14** [comment]
Closing as we've been doing this 😄
